dcicutils/glacier_utils.py

```diff
@@ -0,0 +1,459 @@
+import boto3
+from typing import Union, List, Tuple
+from concurrent.futures import ThreadPoolExecutor
+from tqdm import tqdm
+from .misc_utils import PRINT
+from .ff_utils import get_metadata, search_metadata, get_health_page, patch_metadata
+from .creds_utils import CGAPKeyManager
+
+
+# See https://docs.aws.amazon.com/AmazonS3/latest/userguide/storage-class-intro.html
+# See boto3 docs for info on possible values, but these 3 are the current ones used for
+# glacier (that require restore calls) - Will 7 Apr 2023
+GLACIER_CLASSES = [
+    'GLACIER_IR',  # Glacier Instant Retrieval
+    'GLACIER',  # Glacier Flexible Retrieval
+    'DEEP_ARCHIVE'  # Glacier Deep Archive
+]
+
+
+class GlacierRestoreException(Exception):
+    pass
+
+
+class GlacierUtils:
+
+    def __init__(self, env_name: str):
+        """ Pass an env_name that exists in your ~/.cgap_keys.json or ~/.fourfront_keys.json file
+            No support for admin keys!
+
```

Comment:
It's hard to know without having used these methods yet, but naively I would think some of these methods could be abused, likely unintentionally, to cause issues or unintended effects by an inexperienced user that would result in unnecessary costs/incorrect metadata. I recommend some negative feedback that may be possible in further comments below, but I don't know if they are all possible with the information AWS provides. 

I don't think these are must-haves for now, but consider adding a warning line somewhere to this effect so users can know the lack of guardrails as well as maybe adding some TODOs for possible additions that could prevent unwanted actions.

Reply:
Agreed on "guard rails" generally... Will think on this more